ccm_web/server/src/api/api.service.ts

```diff
@@ -21,4 +29,44 @@ export class APIService {
       }
     }
   }
+
+  async getCourseName (userLoginId: string, courseId: number): Promise<string | null> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId)
+    let name = null
+    try {
+      const response = await requestor.get<Course>(`courses/${courseId}`)
+      const course = response.body
+      name = course.name
+    } catch (error) {
+      if (error instanceof HTTPError && error.response.statusCode !== HttpStatus.OK) {
+        logger.error(`Received unusual status code ${String(error.response.statusCode)}`)
+        logger.error(`Response body: ${JSON.stringify(error.response.body)}`)
+      } else {
+        logger.error(`An error occurred while making a request to Canvas: ${JSON.stringify(error)}`)
+      }
+      throw new Error(error)
+    }
+    return name
+  }
+
+  async putCourseName (userLoginId: string, courseId: number, newName: string): Promise<string | null> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId)
+    let name = null
+    try {
+      const response = await requestor.requestUrl<Course>(
+        `courses/${courseId}`, 'PUT', { course: { name: newName } }
+      )
+      const course = response.body
+      name = course.name
+    } catch (error) {
+      if (error instanceof HTTPError && error.response.statusCode !== HttpStatus.OK) {
+        logger.error(`Received unusual status code ${String(error.response.statusCode)}`)
+        logger.error(`Response body: ${JSON.stringify(error.response.body)}`)
+      } else {
+        logger.error(`An error occurred while making a request to Canvas: ${JSON.stringify(error)}`)
+      }
+      throw new Error(error)
```

Comment:
You have a throw statement here and that will raises `api.controller.ts` and I feel that should  have try-catch block as well. I don't think `result==null` will be hit when you throw an error. 


```
 const result = await this.apiService.putCourseName(userLoginId, courseId, courseNameDto.newName)
    if (result === null) throw new InternalServerErrorException('Error occurred while communicating with Canvas')
    return result
```

I really couldn't test this from swagger UI since from the UI I received different error message

![Screen Shot 2021-06-15 at 4 52 57 PM](https://user-images.githubusercontent.com/8579775/122122916-692d6480-cdfb-11eb-95da-f7a182b7308d.png)

I really don't see the message you are wording to send is the response. So I don't know what going on here

Reply:
Yeah, I think you're right, we don't want to throw the error here, just return null. I'll fix that.

Reply:
Sorry, the error handling has been tricky, it's changed a few times.

Reply:
I removed the error throwing here (https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/121/commits/7ccd58833f005014ad2a5be9af13543491eca96b), but I'm going to see why the specific error string isn't being used that I passed to the `Exception` in the controller.

Reply:
Ah, the message is now showing up now that I'm not throwing an error. This should be resolved. We can discuss more specific error messages in this thread: https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/121#discussion_r652144801